packages/frontpage/app/api/cron/rank-posts/route.ts

```diff
@@ -0,0 +1,51 @@
+import { db } from "@/lib/db";
+import { sendDiscordMessage } from "@/lib/discord";
+import type { NextRequest } from "next/server";
+import * as schema from "@/lib/schema";
+import { sql } from "drizzle-orm";
+
+export async function GET(request: NextRequest) {
+  const authHeader = request.headers.get("authorization");
+  if (authHeader !== `Bearer ${process.env.CRON_SECRET}`) {
+    await sendDiscordMessage({
+      embeds: [
+        {
+          title: "Unauthorized request to cron endpoint",
+          description: `Request: ${request.url}`,
+          color: 16711680,
+        },
+      ],
+    });
+    return new Response("Unauthorized", {
+      status: 401,
+    });
+  }
+
+  try {
+    await db.update(schema.PostAggregates).set({
```

Comment:
Probably worth extracting this (and the same one for comments) into the data layer so it can be called on post creation and comment creation.